### PR TITLE
Feature add distance

### PR DESCRIPTION
**Pull request by tautvilis**

### CODE REVIEW

File: Gariunai Cloud Services/Backend/Coordinates.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        public class Coords
+        {
+            public double Longitude;
+            public double Latitude;
+        }
+
+
+        private static Coords GetCoords(string address)
+        {
+            var mod = "&key=b2fed39c20974a4ca2ef3dbd25cabfe0&pretty=1&limit=1";
```
Comment (accountas) [inline Gariunai Cloud Services/Backend/Coordinates.cs:37]: bad idea to store api keys to github

File: Gariunai Cloud Services/Backend/Coordinates.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        }
+
+
+        private static Coords GetIpLocation()
+        {
+            using var client = new WebClient();
+            client.Headers.Add("user-agent", "Mozilla / 5.0(Windows NT 10.0; WOW64) "
+                                             + "AppleWebKit / 537.36(KHTML, like Gecko)" + "Chrome / 85.0.4183.83 Safari / 537.36");
+            var getExternalIp = client.DownloadString("http://checkip.dyndns.org/");
```
Comment (accountas) [inline Gariunai Cloud Services/Backend/Coordinates.cs:62]: variable naming does not match content, would name it responseString or something similar

File: Gariunai Cloud Services/Backend/Coordinates.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            var removeChars = new Regex("[^0-9.]");
+            var externalIp = removeChars.Replace(getExternalIp, "");
+
+
+            var url = "http://api.ipstack.com/" + externalIp + "?access_key=f16f539429f878f06def52e0e36a81d9";
```
Comment (accountas) [inline Gariunai Cloud Services/Backend/Coordinates.cs:67]: another API key

File: Gariunai Cloud Services/Backend/Coordinates.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                var coords = new double[4];
+                coords[0] = shop.Latitude;
+                coords[1] = shop.Longitude;
+                coords[2] = user.Latitude;
+                coords[3] = user.Longitude;
+
+                var distance = sCoords.GetDistanceTo(uCoords)/1000;
+                Debug.WriteLine(coords);
+
+                //return String.Format("{0:0.00} km",distance);
```
Comment (accountas) [inline Gariunai Cloud Services/Backend/Coordinates.cs:116]: remove unnecessary code

File: Gariunai Cloud Services/LocalProduce/ListShop.cs [hunk 5]
```diff
@@ -40,11 +48,62 @@ public Shop Shop
...
+        {
+            get => _distance;
+            set
+            {
+                _distance = value;
+                ListShopDistance.Text = value;
+            }
+        }
+
+        private class Coords
```
Comment (accountas) [inline Gariunai Cloud Services/LocalProduce/ListShop.cs:61]: why not struct ?

File: Gariunai Cloud Services/SpecificShop/SpecificShopForm.cs [hunk 6]
```diff
@@ -13,17 +12,21 @@ namespace Gariunai_Cloud_Services
...
+        
 
-        public SpecificShopForm(Form previousForm, Shop shop)
+        public SpecificShopForm(Form previousForm, Shop shop, double lng, double lat)
         {
             InitializeComponent();
             _previousForm = previousForm;
             _shop = shop;
+            _latCO = lat;
+            _longCO = lng;
```
Comment (accountas) [inline Gariunai Cloud Services/SpecificShop/SpecificShopForm.cs:29]: why not name ir _latitude and _longitude

**Review by accountas — APPROVED** [review]

**Review by simazm8 — APPROVED** [review]
Martynas made some good points

**Pull request merged**